Realm/Tests/MigrationTests.mm

```diff
@@ -1398,6 +1398,390 @@ - (void)testModifyPrimaryKeyInMigration {
+    RLMObject *secondParentsChild = secondParentObject[@"childProperty"];
+    XCTAssertEqual([secondParentsChild[@"intProperty"] intValue], 43);
+}
+
+- (void)testChangeToEmbeddedWithMultipleBacklinksWithoutProperMigration {
+    RLMProperty *intProperty = [[RLMProperty alloc] initWithName:@"intProperty"
+                                                            type:RLMPropertyTypeInt
+                                                 objectClassName:nil
+                                          linkOriginPropertyName:nil
+                                                         indexed:NO
+                                                        optional:NO];
+    RLMObjectSchema *childObjectSchema = [[RLMObjectSchema alloc] initWithClassName:@"ChildClass"
+                                                                  objectClass:RLMObject.class
+                                                                   properties:@[intProperty]];
+    RLMProperty *childProperty = [[RLMProperty alloc] initWithName:@"childProperty"
+                                                              type:RLMPropertyTypeObject
+                                                   objectClassName:@"ChildClass"
+                                            linkOriginPropertyName:nil
+                                                           indexed:NO
+                                                          optional:YES];
+    RLMObjectSchema *parentObjectSchema = [[RLMObjectSchema alloc] initWithClassName:@"ParentClass"
+                                                                   objectClass:RLMObject.class
+                                                                    properties:@[childProperty]];
+    [self createTestRealmWithSchema:@[parentObjectSchema, childObjectSchema] block:^(RLMRealm *realm) {
+        id childObject = [realm createObject:@"ChildClass" withValue:@[@42]];
+        [realm createObject:@"ParentClass" withValue:@[childObject]];
+        [realm createObject:@"ParentClass" withValue:@[childObject]];
+    }];
+    
+    childObjectSchema.isEmbedded = YES;
+    
+    RLMRealmConfiguration *realmConfiguration = self.config;
+    realmConfiguration.customSchema = [self schemaWithObjects:@[parentObjectSchema, childObjectSchema]];
+    realmConfiguration.schemaVersion = 1;
+    __block int parentEnumerateCalls = 0;
+    __block int childEnumerateCalls = 0;
+    realmConfiguration.migrationBlock = ^(RLMMigration *migration, uint64_t) {
+        [migration enumerateObjects:@"ParentClass" block:^(RLMObject *oldObject, RLMObject *newObject) {
+            parentEnumerateCalls++;
+            XCTAssertNotNil(oldObject);
+            XCTAssertNotNil(newObject);
+            XCTAssertEqual(oldObject[@"childProperty"][@"intProperty"], newObject[@"childProperty"][@"intProperty"]);
+            XCTAssert([newObject[@"childProperty"][@"intProperty"] intValue] == 42 || [newObject[@"childProperty"][@"intProperty"] intValue] == 43);
```

Comment:
This should read the value to a local variable rather than duplicating all that stuff.